src/OpenDiffix.Core/Expression.fs

```diff
@@ -67,6 +67,7 @@ let typeOfAggregate fn args =
   | DiffixCount -> IntegerType
   | DiffixLowCount -> BooleanType
   | Sum -> RealType
+  | DiffixSum -> RealType
```

Comment:
You can peak the first arg to get its type.

Reply:
ok, but two questions here:
1/ is it type stable? (see PR description)
2/ what next? Does it fit under `AggregateOptions` so I can next parametrize `Aggregators.DiffixSum` with it? Or should it rather be a trait of the union member `AggregateFunction.DiffixSum`?

Reply:
Hm, probably answer to 1/ is yes, I'm not sure why I would think otherwise. The issue with the count of AID instances must have sent me off the scent somehow.

Reply:
> can we always know upfront, what arbitrary expression's type will be, before starting to execute?

Yes, we have static type support. If the type can't be resolved to `real` or `integer`, the analyzer should fail to compile.

> so I can next parametrize Aggregators.DiffixSum with it?

Can you inject the required data it in the aggregator constructor? Seems like the most straightforward solution.

Reply:
> Can you inject the required data it in the aggregator constructor? Seems like the most straightforward solution.

ok, lemme try this and come back to you with a specific question (if any). 

Thanks for the clarification, didn't think about the analyzer failure part.

Reply:
You are right, sorry - I thought we had them in scope.

> I'm now working on including the AggregateArgs in Aggregator.create, but it's a bumpy road

Shouldn't be too difficult, I can make a patch if you like.

Reply:
I'm close, but not quite there yet. Let me give it some more tries, and we can sit down together next week, if I get stuck

Reply:
OK, I've got this, thank you!

Now, I'm tempted to simplify the initialization of the Aggregator classes leveraging the `aggArgs` we now have readily available (this not being done was what made me think typed aggregator is not possible in the first place). WDYT?

Reply:
> OK, I've got this, thank you!
> 
> Now, I'm tempted to simplify the initialization of the Aggregator classes leveraging the `aggArgs` we now have readily available (this not being done was what made me think typed aggregator is not possible in the first place). WDYT?
> 
> (if at all, in a follow-up PR, of course)

Sorry, I missed this question. By now I have lost context of what we were doing, but I'm OK with simplifying things.

Reply:
ha, and I've missed your answer. I can make the follow up PR to demonstrate what I'm getting at.